mindmeld/components/entity_resolver.py

```diff
@@ -569,31 +738,242 @@ def _predict_exact_match(self, entity):
+                .get("pretrained_name_or_abspath", "bert-base-nli-mean-tokens")
+        )
+        self._sbert_model = None
+
+        # TODO: _lazy_resolution is set to a default value, can be modified to be an input
+        self._lazy_resolution = False
+        if not self._lazy_resolution:
+            msg = "sentence-bert embeddings are cached for entity_type: {%s} " \
+                  "for fast entity resolution; can possibly consume more disk space"
+            logger.warning(msg, self.type)
+
+    @property
+    def pretrained_name(self):
+        return os.path.split(self._sbert_model_pretrained_name_or_abspath)[-1]
+
+    def _encode(self, phrases):
+        """Encodes input text(s) into embeddings, one vector for each phrase
+
+        Args:
+            phrases (str, list[str]): textual inputs that are to be encoded using sentence \
+                                        transformers' model
+
+        Returns:
+            list[np.array]: one numpy array of embeddings for each phrase,
+                            if ``phrases`` is ``str``, a list of one numpy aray is returned
+        """
+
+        if not phrases:
+            return []
+
+        if isinstance(phrases, str):
+            phrases = [phrases]
+
+        if not isinstance(phrases, (str, list)):
+            raise TypeError(f"argument phrases must be of type str or list, not {type(phrases)}")
+
+        # batch_size (int):
+        #   The maximum size of each batch while encoding using on a deep embedder like BERT
+        _batch_size = (
+            self.er_config.get("model_settings", {})
+                .get("batch_size", 16)
+        )
+        show_progress = len(phrases) > 1
+        return self._sbert_model.encode(phrases,
+                                        batch_size=_batch_size,
+                                        convert_to_numpy=True,
+                                        show_progress_bar=show_progress)
+
+    @staticmethod
+    def _sort_using_cosine_dist(syn_embs, entity_emb):
+        """Uses cosine similarity metric on synonym embeddings to sort most relevant ones
+            for entity resolution
+
+        Args:
+            syn_embs (list[np.array]): a list of synonym embeddings
+            entity_emb: (np.array): embedding of input entity
+        Returns:
+            list[tuple]: a list of sorted synonym names, paired with their similarity scores \
+                        (descending)
+        """
+
+        entity_emb = entity_emb.reshape(1, -1)
+        synonyms, synonyms_encodings = zip(*syn_embs.items())
+        similarity_scores = cosine_similarity(np.array(synonyms_encodings), entity_emb).reshape(-1)
+        similarity_scores = np.around(similarity_scores, decimals=2)
+
+        # results in descending scores
+        return sorted(list(zip(synonyms, similarity_scores)), key=lambda x: x[1], reverse=True)
+
+    @staticmethod
+    def _process_entity_map(entity_type, entity_map):
```

Comment:
I think a lot of the logic in this function and `EntityResolverUsingExactMatch `'s function can be consolidated

Reply:
Consolidated into a static method in base class in the latest PR